File: Minimum Moves to Equal Array Elements II.py
#https://leetcode.com/problems/minimum-moves-to-equal-array-elements-ii/description/?envType=problem-list-v2&envId=sorting

class Solution:
    def minMoves2(self, nums: List[int]) -> int:
        nums=sorted(nums)
        minn=nums[0]
        maxx=nums[len(nums)-1]
        count=0
        result=10**10+7
        for j in nums[int(len(nums)/2)-1:int(len(nums)/2)+1]:
            for i in nums:
                count+=abs(i-j)
            result=min(result,count)
            count=0
        return int(result)

File: Minimum Moves to Equal Array Elements II.cpp
//https://leetcode.com/problems/minimum-moves-to-equal-array-elements-ii/description/


class Solution {
public:
    int minMoves2(vector<int>& nums) {
        sort(nums.begin(),nums.end());
        int median=nums[nums.size()/2];

        int total=0;

        for(int i:nums){
            total+=abs(i-median);
        }
        return total;
    }
};
